sources/zafira-web/src/main/java/com/qaprosoft/zafira/web/FilterController.java

```diff
@@ -47,63 +40,45 @@ public FilterController(FilterService filterService, Mapper mapper) {
     }
 
     @ApiResponseStatuses
-    @ApiOperation(value = "Create filter", nickname = "createFilter", httpMethod = "POST", response = FilterType.class)
-    @ApiImplicitParams({ @ApiImplicitParam(name = "Authorization", paramType = "header") })
+    @ApiOperation(value = "Create filter", nickname = "createFilter", httpMethod = "POST", response = FilterDTO.class)
+    @ApiImplicitParams({@ApiImplicitParam(name = "Authorization", paramType = "header")})
     @PostMapping()
-    public FilterType createFilter(@RequestBody @Valid FilterType filterType) {
-        filterType.setUserId(getPrincipalId());
-        filterType.getSubject().sortCriterias();
-
-        Filter filter = mapper.map(filterType, Filter.class);
-        if (filterService.isFilterExists(filter)) {
-            throw new IllegalOperationException(FILTER_CAN_NOT_BE_CREATED, ERR_MSG_FILTER_WITH_SUCH_NAME_ALREADY_EXISTS);
-        }
-        if (filter.isPublicAccess() && !isAdmin()) {
-            filter.setPublicAccess(false);
-        }
-
-        return mapper.map(filterService.createFilter(filter), FilterType.class);
+    public FilterDTO createFilter(@RequestBody @Valid FilterDTO filterDTO) {
+        filterDTO.getSubject().sortCriterias();
+        return mapper.map(filterService.createFilter(mapper.map(filterDTO, Filter.class), getPrincipalId(), isAdmin()), FilterDTO.class);
```

Comment:
Let's shift from Bogdan-style chain calls to more readable constructions - introduce a local variable for every statement executed within this line